src/plugins/FileService/MainDialog.tsx

```diff
@@ -0,0 +1,105 @@
+import { DialogContent, DialogProps, DialogTitle, Grid, IconButton, makeStyles, Typography } from '@material-ui/core'
+import { isNil } from 'lodash-es'
+import { useSnackbar } from 'notistack'
+import React from 'react'
+import { useStylesExtends } from '../../components/custom-ui-helper'
+import { DialogDismissIconUI } from '../../components/InjectedComponents/DialogDismissIcon'
+import { getActivatedUI } from '../../social-network/ui'
+import { useI18N } from '../../utils/i18n-next-ui'
+import ShadowRootDialog from '../../utils/jss/ShadowRootDialog'
+import { Entry } from './components'
+import { InsertButton } from './components/InsertButton'
+import { META_KEY_1 } from './constants'
+import { Exchange } from './hooks/Exchange'
+import type { FileInfo } from './types'
+
+interface Props
+    extends withClasses<
+        | KeysInferFromUseStyles<typeof useStyles>
+        | 'dialog'
+        | 'backdrop'
+        | 'container'
+        | 'paper'
+        | 'input'
+        | 'header'
+        | 'content'
+        | 'close'
+        | 'button'
+        | 'label'
+        | 'switch'
+    > {
+    open: boolean
+    onConfirm: (file: FileInfo | undefined) => void
+    onDecline: () => void
+    DialogProps?: Partial<DialogProps>
+}
+
+const useStyles = makeStyles({
+    title: { marginLeft: 6 },
+    container: { width: '100%' },
+    content: { padding: 12, background: '#fff' },
+})
+
+const MainDialog: React.FC<Props> = (props) => {
+    const { t } = useI18N()
+    const classes = useStylesExtends(useStyles(), props)
+    const snackbar = useSnackbar()
+    const [uploading, onUploading] = React.useState(false)
```

Comment:
Better to use `setUploading`, it's easy to understand it as an event callback function.